bayesian/utils.py

```diff
@@ -382,8 +382,9 @@ def get_analyses_for_snyk(self):
             if "cve" in result_data[0]:
                 # if cve if present
                 clubbed_data.append({
-                    "recommended_versions": result_data[0]['cve']['sfixed_in'],
-                    "link": self.get_link(),
+                    "recommended_versions": result_data[0].get('package', {}).get(
```

Comment:
have a if condition here to get the data from latest_non_cve_version, if not found, send blank and add a logger.error msg. Dont add exception. just an error msg

Reply:
something like this
rec_version = ""
if "latest_non_cve_version" in result_data[0]['package']:
 rec_version = bla bla
else
log msg
clubbed_data.appen ........

Reply:
@yzainee Logic covers missing `latest_non_cve_version` field.
I mean 
```
"recommended_versions": result_data[0].get('package', {}).get(
                                                        'latest_non_cve_version', {}),
```
will return `{}` if there is missing `latest_non_cve_version`

Reply:
i know the logic covers that. i only want a msg in logs for such pkgs so that investigation can be done if needed. right now we wont know why it returned blank

Reply:
whichever way you want, just add a msg if you dont find this value.. in msg just have eco pkg. should be enough

Reply:
or if you dont want to add an if condition, you can add a generic logger.info("latest non cve version for {eco} {pkg} is {ver}") . then whether +ve or -ve, we will see this msg in logs

Reply:
Sure. I am actually reluctant in adding any conditional statement here... as it will Increase Cyclomatic complexity,which hence make unit Testing a bit harder and may cause Coverage to reduce.

Sure. Going ahead with adding Logs